packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -238,6 +243,33 @@ class LinkViewController extends PlatformViewController {
     _element.setAttribute('target', _getHtmlTarget(target));
   }
 
+  /// Set the draggable value for this link.
+  void setDraggable(bool draggable) {
+    assert(_isInitialized);
+    if (!draggable) {
+      _element.style
+        // Chrome
+        ..userDrag = 'none'
+        // Edge
+        ..setProperty('-webkit-user-drag', 'none');
+      // Firefox
+      _element.draggable = false;
+      _dragSubscription = _element.onDragStart.listen((event) {
+        event.preventDefault();
+      });
```

Comment:
I was checking caniuse.com, and it seems that:

* there's no unprefixed `user-drag` property. [Docs](https://caniuse.com/webkit-user-drag).
* wouldn't `draggable = false` be enough?

Also, if disabling drag in the element, is it necessary to stop the default drag event?

I think this method could be much simplified to be just `_element.draggable = false`?

Reply:
It seems you're close to being right. We started out with the styling approach since it allowed us to apply a global work-around in index.html (`<style>a { -webkit-user-drag: none; }</style>`). Probably just added `user-drag` since it was right there in `html_dart2js.dart`, and if it works, it's nicer than the `-webkit-` style. But it also seems that `draggable` works on both Chrome and Edge; however, I couldn't make it work on Firefox without `preventDefault`. In the end, I guess it's a matter of taste: Do you want style or properties for this? Do you want to wrap the drag subscription in some Firefox shim?

Reply:
After some tests, it seems that **the only solution that seems to work the same across all platforms is `preventDefault` on the `dragstart` event.** See:

* https://jsfiddle.net/ditman/2kLuo8vd

If `draggable=false`, none of the browsers will trigger a `dragstart` event (as expected).

In Firefox, `draggable=false` works (it won't let you drag the Link), **but** it'll also start selecting from the text of the link as the user drags (which I guess is what you couldn't get to work?). This can be "corrected" with `user-select`, but then the Link won't be highlighted at all when text is selected from the outside. This looks like a Firefox bug/feature, but I couldn't find anything relevant in [their bug tracker](https://bugzilla.mozilla.org/buglist.cgi?bug_status=__open__&component=DOM%3A%20Selection&product=Core&query_format=advanced&order=bug_status%2Cpriority%2Cassigned_to%2Cbug_id&limit=0).

Reply:
(My second favorite version might be `draggable=false` and if the browser is firefox, prevent the `selectstart` event (`draggable=false + no selectstart` in the fiddle), which doesn't interfere with the selection from the outside as the `user-select` CSS attribute does!)

Reply:
Actually, what we couldn't get to work is [the issue I linked to](https://github.com/flutter/flutter/issues/102735): links in a list. For that case preventing select start doesn't seem to be sufficient. But interactions with text selection should indeed probably also be considered.